clippy_lints/src/anon_trait_import.rs

```diff
@@ -0,0 +1,72 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet_opt;
+use rustc_errors::Applicability;
+use rustc_hir::def::{DefKind, Res};
+use rustc_hir::{Item, ItemKind, UseKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::Visibility;
+use rustc_session::declare_lint_pass;
+use rustc_span::symbol::kw;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `use Trait` where the Trait is not used directly.
+    ///
+    /// ### Why is this bad?
+    /// Traits imported that aren't used directly can be imported anonymously `use Trait as _`. It is more explicit and can be useful to show which imports are required for traits.
+    ///
+    /// ### Example
+    /// ```no_run
+    ///   use std::fmt::Write;
```

Comment:
Can you remove the extra indentation from the examples here? The rendered markdown currently looks like this
![image](https://github.com/user-attachments/assets/7c0099a9-1fa1-46b9-b1cf-d5311fe087cb)